mcs/class/WindowsBase/ZipSharp/UnzipReadStream.cs

```diff
@@ -30,6 +30,7 @@ namespace zipsharp
 {
 	class UnzipReadStream : Stream
 	{
+		private static bool use32Bit = IntPtr.Size == 4 || Environment.OSVersion.Platform != PlatformID.Unix;
```

Comment:
I am not sure I'd split the logic as some methods have different parameter type and other don't